web/packages/teleterm/src/ui/DocumentCluster/UnifiedResources.tsx

```diff
@@ -0,0 +1,287 @@
+                    <ConnectDatabaseActionButton database={database} />
+                  ),
+                },
+              };
+            }
+            case 'kube': {
+              const { resource: kube } = resource;
+
+              return {
+                resource: {
+                  kind: 'kube_cluster' as const,
+                  labels: kube.labelsList,
+                  name: kube.name,
+                },
+                ui: {
+                  ActionButton: <ConnectKubeActionButton kube={kube} />,
+                },
+              };
+            }
+          }
+        })
+        .filter(Boolean)}
+      resourcesFetchAttempt={attempt}
+      fetchResources={fetch}
+      availableKinds={['db', 'kube_cluster', 'node']}
+      Header={pinAllButton => (
+        <Flex alignItems="center" justifyContent="space-between">
+          {/*temporary search panel*/}
+          <SearchPanel
+            params={params}
+            pathname={''}
+            replaceHistory={() => undefined}
+            setParams={setParams}
+          />
+          {pinAllButton}
+        </Flex>
+      )}
+      NoResources={
+        <NoResources
+          canCreate={canAddResources}
+          canUseConnectMyComputer={canUseConnectMyComputer}
+          onConnectMyComputerCtaClick={() => {
+            documentsService.openConnectMyComputerDocument({ rootClusterUri });
+          }}
+        />
+      }
+    />
+  );
+}
+
+interface NoResourcesProps {
+  canCreate: boolean;
+  canUseConnectMyComputer: boolean;
+  onConnectMyComputerCtaClick(): void;
+}
+
+function NoResources(props: NoResourcesProps) {
+  let $content: React.ReactElement;
+  if (!props.canCreate) {
+    $content = (
+      <>
+        <Text typography="h3" mb="2" fontWeight={600}>
+          No Resources Found
+        </Text>
+        <Text>
+          Either there are no resources in the cluster, or your roles don't
+          grant you access.
+        </Text>
+      </>
+    );
+  } else {
+    $content = (
+      <>
+        <Image src={stack} ml="auto" mr="auto" mb={4} height="100px" />
+        <Text typography="h3" mb={2} fontWeight={600}>
+          Add your first resource to Teleport
+        </Text>
+        <Text color="text.slightlyMuted">
+          {props.canUseConnectMyComputer
+            ? 'You can add it in the Teleport Web UI or by connecting your computer to the cluster.'
+            : 'Connect SSH servers, Kubernetes clusters, Databases and more from Teleport Web UI.'}
+        </Text>
+        {props.canUseConnectMyComputer && (
+          <ButtonPrimary
+            type="button"
+            mt={3}
+            gap={2}
+            onClick={props.onConnectMyComputerCtaClick}
+          >
+            <icons.Laptop size={'medium'} />
+            Connect My Computer
+          </ButtonPrimary>
+        )}
+      </>
+    );
+  }
+
+  return (
+    <Box
+      maxWidth={600}
+      p={8}
+      pt={5}
+      as={Flex}
```

Comment:
Why not use `Flex` in the first place?

Reply:
I think I copied it from somewhere else, changed to `Flex`.